subsys/bluetooth/host/hci_core.c

```diff
@@ -3650,6 +3659,15 @@ static void le_adv_report(struct net_buf *buf)
 			net_buf_simple_restore(&buf->b, &state);
 		}
 
+		SYS_SLIST_FOR_EACH_CONTAINER(&scan_cbs, cb, node) {
```

Comment:
I realized a problem when including this change in my branch.
scan_dev_found_cb is set back to NULL when scanner is stopped, this keeps it from reporting on packets received when using scanner before establishing connection.

But this part will continue to report packets during background scan. I would think we do don't want that.
Should be:
```
if (atomic_test_bit(bt_dev.flags, BT_DEV_EXPLICIT_SCAN)) {
    SYS_SLIST_FOR_EACH_CONTAINER(&scan_cbs, cb, node) {
    ...
    }
}

Reply:
After #21717, this won't happen anymore, but even if it did, I'd argue that this is actually the desired behavior. Contrary to the callback passed to the scan_start API, this callback doesn't live in the `EXPLICIT_SCAN` context, and should report on any activity observed by the scanner, regardless of what state it was in.

Reply:
@trond-snekvik Fair point. It should be documented in that case.